From 16ed9086dccf79bc1f46f604730e538da501d4c6 Mon Sep 17 00:00:00 2001
From: czietz <czietz@gmx.net>
Date: Sat, 22 Sep 2018 17:13:45 +0100
Subject: [PATCH 1/3] Enable SCC interrupt in the System Control Unit.

---
 bios/serport.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/bios/serport.c b/bios/serport.c
index 30afada2..a7a16c19 100644
--- a/bios/serport.c
+++ b/bios/serport.c
@@ -557,6 +557,15 @@ ULONG reset_recovery_loops;
     /* initialise channel B */
     for (p = SCC_init_string; *p >= 0; p++)
         write_scc(&scc->portB,HIBYTE(*p),LOBYTE(*p));
+
+    /*
+     * Enable routing of the SCC interrupt through the SCU like TOS does.
+     * Even though interrupts are not used here, other programs might
+     * install their own interrupt vectors and expect the interrupt
+     * to be available to them.
+     */
+     if (HAS_VME)
+        *(volatile BYTE *)VME_INT_MASK |= 1<<5;
 }
 #endif  /* CONF_WITH_SCC */
 
-- 
2.17.0.windows.1

