From accc009af03a5770e44dce879873fff4a1f3f273 Mon Sep 17 00:00:00 2001
From: czietz <czietz@gmx.net>
Date: Fri, 1 Mar 2019 20:52:07 +0000
Subject: [PATCH] Optimize muls() inline function

---
 include/intmath.h | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/include/intmath.h b/include/intmath.h
index fe413b3f..1c366d32 100644
--- a/include/intmath.h
+++ b/include/intmath.h
@@ -66,10 +66,9 @@ static __inline LONG muls(WORD m1, WORD m2)
     LONG ret;
 
     __asm__ (
-      "muls %2,%0\n\t"
-      "move.l %0,%1"
-    : "+d"(m1), "=idm" (ret)
-    : "idm"(m2)
+      "muls %2,%0"
+    : "=d"(ret)
+    : "%0"(m1), "idm"(m2)
     );
 
     return ret;
-- 
2.17.0.windows.1

