From 3db6c1ce157ffd1e2fde9f573ecc7fdfa670f638 Mon Sep 17 00:00:00 2001
From: Christian Zietz <czietz@gmx.net>
Date: Sun, 7 May 2017 14:01:49 +0100
Subject: [PATCH 1/9] Remove unused win_iname function

---
 desk/deskwin.c | 18 ------------------
 1 file changed, 18 deletions(-)

diff --git a/desk/deskwin.c b/desk/deskwin.c
index 38f5aa5..536e50f 100644
--- a/desk/deskwin.c
+++ b/desk/deskwin.c
@@ -593,24 +593,6 @@ WORD win_isel(OBJECT olist[], WORD root, WORD curr)
 
 
 /*
- *  Return pointer to this icon's name.  It will always be an icon that
- *  is on the desktop.
- */
-BYTE *win_iname(WORD curr)
-{
-    ICONBLK *pib;
-    BYTE    *ptext;
-
-    assert(G.g_screen[curr].ob_type == G_ICON);
-
-    pib = (ICONBLK *)G.g_screen[curr].ob_spec;
-    ptext = pib->ib_ptext;
-
-    return ptext;
-}
-
-
-/*
  *  Set the name and information lines of a particular window
  */
 void win_sname(WNODE *pw)
-- 
2.8.2.windows.1

