From 84dd7c1087067ba9255c7b1eb0d9536dff4c97e8 Mon Sep 17 00:00:00 2001
From: Thorsten Otto <thotto@users.sourceforge.net>
Date: Wed, 19 Apr 2017 14:44:52 +0200
Subject: [PATCH] Make check_read_byte() work on ARAnyM-JIT

---
 bios/vectors.S | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/bios/vectors.S b/bios/vectors.S
index 83fb66d..be1eba7 100644
--- a/bios/vectors.S
+++ b/bios/vectors.S
@@ -1161,13 +1161,18 @@ _check_read_byte:
         moveq   #0,d0
         nop                                     // flush pipeline
 
+        jsr     dummy_jit_workaround
         move.l  4(sp),a0
+        jsr     dummy_jit_workaround
         tst.b   (a0)
+        jsr     dummy_jit_workaround
         nop                                     // flush pipeline
 
         moveq   #0x01,d0        // passed
 
-berr:   move.l  a1,(8).w
+berr:   
+        jsr     dummy_jit_workaround
+        move.l  a1,(8).w
         move.l  d1,sp
         nop
 
@@ -1197,6 +1202,7 @@ msgcrb:
  * _just_rts - just a dummy
  */
 
+dummy_jit_workaround:
 _just_rts:
         rts
 
-- 
2.1.2

