From e8234ae84d1ee8210f54b6eab92fc87e24bccfe4 Mon Sep 17 00:00:00 2001
From: Christian Zietz <czietz@gmx.net>
Date: Mon, 19 Oct 2020 16:36:31 +0100
Subject: [PATCH] Just one global IDE timeout

---
 bios/ide.c | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/bios/ide.c b/bios/ide.c
index c5e55f74..0fd6bc1c 100644
--- a/bios/ide.c
+++ b/bios/ide.c
@@ -392,24 +392,18 @@ enum ide_if_status
  *    b. no device is found on both regular and twisted interfaces
  *    c. a timeout occurs, i.e., both interfaces stayed BSY
  */
-static int ide_interface_exists(WORD ifnum)
+static int ide_interface_exists(WORD ifnum, LONG timeout)
 {
     volatile struct IDE *regular_iface = ifinfo[ifnum].base_address;
     volatile struct IDE *twisted_iface = (volatile struct IDE *)(((ULONG)ifinfo[ifnum].base_address)-1);
     enum ide_if_status  regular_iface_status = IDE_IF_NOTCHECKED;
     enum ide_if_status  twisted_iface_status = IDE_IF_NOTCHECKED;
 
-    /* We wait a max time for BSY to drop since this is called during
-     * initialisation, which can be invoked by power-on/reset.
-     */
-    LONG next = hz_200 + LONG_TIMEOUT;
-
     IDE_WRITE_CONTROL(regular_iface,IDE_CONTROL_nIEN);/* no interrupts please */
     IDE_WRITE_CONTROL(twisted_iface,IDE_CONTROL_nIEN);/* no interrupts please */
 
     DELAY_400NS;
-    while ((hz_200 < next) &&
-        ((regular_iface_status == IDE_IF_NOTCHECKED) || (twisted_iface_status == IDE_IF_NOTCHECKED))) {
+    do {
         /* Check BSY on regular interface. */
         if ((IDE_READ_ALT_STATUS(regular_iface) & IDE_STATUS_BSY) == 0) {
             /* Check it exists by setting and reading back magic number. */
@@ -446,7 +440,8 @@ static int ide_interface_exists(WORD ifnum)
                 twisted_iface_status = IDE_IF_NOTPRESENT;
             }
         }
-    }
+    } while ((hz_200 < timeout) &&
+             ((regular_iface_status == IDE_IF_NOTCHECKED) || (twisted_iface_status == IDE_IF_NOTCHECKED)));
 
     int rc = (regular_iface_status == IDE_IF_PRESENT) || (twisted_iface_status == IDE_IF_PRESENT);
     KDEBUG(("ide interface %d %s %s\n",ifnum,rc?"exists":"not present",ifinfo[ifnum].twisted_cable?"(twisted cable)":""));
@@ -519,10 +514,15 @@ void ide_init(void)
         return;
 
 #if CONF_ATARI_HARDWARE && !defined(MACHINE_FIREBEE)
-    /* reject 'ghost' interfaces & detect twisted cables */
+    /* Reject 'ghost' interfaces & detect twisted cables.
+     * We wait a max time for BSY to drop on all IDE interface
+     * since this is called during initialisation, which can be
+     * invoked by power-on/reset.
+     */
+    LONG timeout = hz_200 + LONG_TIMEOUT;
     for (i = 0, bitmask = 1; i < NUM_IDE_INTERFACES; i++, bitmask <<= 1)
         if (has_ide&bitmask)
-            if (!ide_interface_exists(i))
+            if (!ide_interface_exists(i, timeout))
                 has_ide &= ~bitmask;
 
     KDEBUG(("ide_init(): has_ide = 0x%02x\n",has_ide));
-- 
2.17.0.windows.1

