From 3f4a83d8ab4b7977fa92e4bc012116246d207109 Mon Sep 17 00:00:00 2001
From: Christian Zietz <czietz@gmx.net>
Date: Wed, 19 Jul 2017 19:26:20 +0100
Subject: [PATCH] Do not use stack until memory is initialized

---
 bios/memory.S | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/bios/memory.S b/bios/memory.S
index 806b736..64c083b 100644
--- a/bios/memory.S
+++ b/bios/memory.S
@@ -69,13 +69,8 @@ meminit:
 
 #if CONF_WITH_FALCON_MMU
         /* detect if this is a Falcon by reading $ff8282 register */
-        pea     0xffff8282.w
-        jsr     _check_read_byte
-        addq.l  #4, sp
-
-        tst.w   d0                      // is it Falcon MMU?
-        jeq     not_a_falcon            // no -> skip 0xffff8007 test
-
+        move.l  #not_a_falcon,0x08
+        tst.b   0xffff8282.w
         bset    #MEMINIT_BIT_FALCON_MMU, d7
 
 #if !EMUTOS_LIVES_IN_RAM
-- 
2.8.2.windows.1

