From ce2706c74447905439125ec791054770a1ef3a60 Mon Sep 17 00:00:00 2001
From: Christian Zietz <czietz@gmx.net>
Date: Sun, 7 May 2017 14:02:17 +0100
Subject: [PATCH 2/9] Remove unused xsuper global symbol

---
 bdos/rwa.S | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/bdos/rwa.S b/bdos/rwa.S
index a304c1c..b131a7e 100644
--- a/bdos/rwa.S
+++ b/bdos/rwa.S
@@ -126,7 +126,6 @@
         .globl  _enter
         .globl  _old_trap2
         .globl  _bdos_trap2
-        .globl  _xsuper
 
 // ==== References ===========================================================
 
@@ -290,13 +289,6 @@ fstrt:
 
 //==== function 0x20 handler - toggle/inquire processor/stack state ==========
 
-//msg_super:
-//      .ascii "BDOS: xsuper - switching to supervisor mode ...\n\0"
-//      .even
-_xsuper:
-//      pea msg_super
-//      bsr _kprint
-//      addq #4,sp
 x20_usr:                        // user mode entry point
         move.l  2(a0),d1        // get parameter
         beq     xu_0            // already have new SSP from
-- 
2.8.2.windows.1

