From f13a93ac2a8e294bdd8cc23927af47ce66b47425 Mon Sep 17 00:00:00 2001
From: Christian Zietz <czietz@gmx.net>
Date: Sat, 14 Apr 2018 12:24:02 +0100
Subject: [PATCH 2/3] New config option CONF_WITH_NOVA

---
 include/config.h | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/include/config.h b/include/config.h
index a84e7bc..dd001ff 100644
--- a/include/config.h
+++ b/include/config.h
@@ -90,6 +90,9 @@
 # ifndef CONF_WITH_MONSTER
 #  define CONF_WITH_MONSTER 0
 # endif
+# ifndef CONF_WITH_NOVA
+#  define CONF_WITH_NOVA 0
+# endif
 # ifndef CONF_WITH_FORMAT
 #  define CONF_WITH_FORMAT 0
 # endif
@@ -156,6 +159,9 @@
 # ifndef CONF_WITH_MONSTER
 #  define CONF_WITH_MONSTER 0
 # endif
+# ifndef CONF_WITH_NOVA
+#  define CONF_WITH_NOVA 0
+# endif
 # ifndef CONF_WITH_FORMAT
 #  define CONF_WITH_FORMAT 0
 # endif
@@ -194,6 +200,9 @@
 # ifndef CONF_WITH_MONSTER
 #  define CONF_WITH_MONSTER 0
 # endif
+# ifndef CONF_WITH_NOVA
+#  define CONF_WITH_NOVA 0
+# endif
 # ifndef CONF_WITH_TTRAM
 #  define CONF_WITH_TTRAM 0
 # endif
@@ -333,6 +342,9 @@
 # ifndef CONF_WITH_MONSTER
 #  define CONF_WITH_MONSTER 0
 # endif
+# ifndef CONF_WITH_NOVA
+#  define CONF_WITH_NOVA 0
+# endif
 #endif
 
 /*
@@ -523,6 +535,9 @@
 # ifndef CONF_WITH_MONSTER
 #  define CONF_WITH_MONSTER 0
 # endif
+# ifndef CONF_WITH_NOVA
+#  define CONF_WITH_NOVA 0
+# endif
 #endif
 
 /*
@@ -923,6 +938,13 @@
 #endif
 
 /*
+ * Set CONF_WITH_NOVA to 1 to enable support for Nova graphic card adapter.
+ */
+#ifndef CONF_WITH_NOVA
+# define CONF_WITH_NOVA 1
+#endif
+
+/*
  * CONF_LOGSEC_SIZE defines the maximum size of logical sectors that
  * GEMDOS can handle, specified in bytes; however, if a larger value
  * is found on a mounted drive during startup, that value will be used
-- 
2.8.2.windows.1

