From 376e3222ef198e19d80850025bd8766d8adf58cb Mon Sep 17 00:00:00 2001
From: Christian Zietz <czietz@gmx.net>
Date: Sun, 7 May 2017 14:02:45 +0100
Subject: [PATCH 3/9] Remove unused shel_find binding

---
 desk/gembind.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/desk/gembind.c b/desk/gembind.c
index 21d6b78..57dff97 100644
--- a/desk/gembind.c
+++ b/desk/gembind.c
@@ -941,11 +941,13 @@ WORD shel_put(const void *pdata, WORD len)
 }
 
 
+/* unused
 WORD shel_find(BYTE *ppath)
 {
     SH_PATH = (LONG)ppath;
     return gem_if(AES_CTRL_CODE(SHEL_FIND, 0, 1, 1));
 }
+*/
 
 
 /* unused
-- 
2.8.2.windows.1

