From bcfe733c2482f1fa61d9a4f9c120045f4176fe5f Mon Sep 17 00:00:00 2001
From: czietz <czietz@gmx.net>
Date: Mon, 18 Feb 2019 18:33:24 +0000
Subject: [PATCH 4/4] Remove call to win_find() that has no side effects

---
 desk/deskmain.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/desk/deskmain.c b/desk/deskmain.c
index 1045464a..4ebcf2b3 100644
--- a/desk/deskmain.c
+++ b/desk/deskmain.c
@@ -642,7 +642,7 @@ static WORD hndl_button(WORD clicks, WORD mx, WORD my, WORD button, WORD keystat
 
     if (clicks == 1)
     {
-        WNODE *pw = win_find(wh);
+        WNODE *pw;
 
         act_bsclick(G.g_cwin, G.g_screen, G.g_croot, mx, my,
                     keystate, &c, FALSE);
-- 
2.17.0.windows.1

