From e45b46e71ad27b130dff627e989b50c0d8e734bc Mon Sep 17 00:00:00 2001
From: Christian Zietz <czietz@gmx.net>
Date: Sun, 7 May 2017 15:37:59 +0100
Subject: [PATCH 5/9] kpgm_relocate is only called with DETECT_NATIVE_FEATURES
 configuration

---
 bdos/kpgmld.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/bdos/kpgmld.c b/bdos/kpgmld.c
index fd42521..1e13c56 100644
--- a/bdos/kpgmld.c
+++ b/bdos/kpgmld.c
@@ -281,6 +281,7 @@ static LONG pgfix01(void *lastcp, LONG nrelbytes, PGMINFO *pi)
 }
 
 
+#if DETECT_NATIVE_FEATURES
 LONG kpgm_relocate(PD *p, long length)
 {
     BYTE    *cp;
@@ -362,3 +363,4 @@ LONG kpgm_relocate(PD *p, long length)
 
     return 0;
 }
+#endif
-- 
2.8.2.windows.1

