From 3256817b43a9ca2dc634f120ac74947bdae2f406 Mon Sep 17 00:00:00 2001
From: Christian Zietz <czietz@gmx.net>
Date: Sun, 7 May 2017 15:41:38 +0100
Subject: [PATCH 7/9] disk_inquire is only needed with CONF_WITH_XHDI

---
 bios/disk.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/bios/disk.c b/bios/disk.c
index e8a4d6e..c616b24 100644
--- a/bios/disk.c
+++ b/bios/disk.c
@@ -744,6 +744,7 @@ static LONG internal_inquire(UWORD unit, ULONG *blocksize, ULONG *deviceflags, c
     return 0;
 }
 
+#if CONF_WITH_XHDI
 /* Get unit information, whatever low-level driver is used. */
 LONG disk_inquire(UWORD unit, ULONG *blocksize, ULONG *deviceflags, char *productname, UWORD stringlen)
 {
@@ -755,6 +756,7 @@ LONG disk_inquire(UWORD unit, ULONG *blocksize, ULONG *deviceflags, char *produc
 
     return internal_inquire(unit, blocksize, deviceflags, productname, stringlen);
 }
+#endif /* CONF_WITH_XHDI */
 
 /* Get unit capacity */
 LONG disk_get_capacity(UWORD unit, ULONG *blocks, ULONG *blocksize)
-- 
2.8.2.windows.1

