From 7d28c0fa07e0f4fd659f2a3aebcc46735c84cb67 Mon Sep 17 00:00:00 2001
From: Christian Zietz <czietz@gmx.net>
Date: Sun, 7 May 2017 15:42:10 +0100
Subject: [PATCH 8/9] pro_exit is only needed with CONF_WITH_SHUTDOWN

---
 desk/deskpro.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/desk/deskpro.c b/desk/deskpro.c
index 76ff7a9..3aab8be 100644
--- a/desk/deskpro.c
+++ b/desk/deskpro.c
@@ -70,6 +70,7 @@ WORD pro_run(WORD isgraf, WORD isover, WORD wh, WORD curr)
 }
 
 
+#if CONF_WITH_SHUTDOWN
 WORD pro_exit(BYTE *pcmd, BYTE *ptail)
 {
     WORD ret;
@@ -77,3 +78,4 @@ WORD pro_exit(BYTE *pcmd, BYTE *ptail)
     ret = shel_write(0, FALSE, 1, pcmd, ptail);
     return ret;
 }
+#endif
-- 
2.8.2.windows.1

