From 81fec70f8bcaa4a5e5a9fb429d2fae34e4c5719e Mon Sep 17 00:00:00 2001
From: Christian Zietz <czietz@gmx.net>
Date: Sun, 7 May 2017 15:43:19 +0100
Subject: [PATCH 9/9] dos_conws and dos_conout are only needed with
 CONF_WITH_SHOW_FILE

---
 util/gemdos.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/util/gemdos.c b/util/gemdos.c
index 028dbd1..70e9db6 100644
--- a/util/gemdos.c
+++ b/util/gemdos.c
@@ -121,10 +121,12 @@ void dos_func(UWORD function, LONG parm)
 */
 
 
+#if CONF_WITH_SHOW_FILE
 void dos_conout(WORD ch)
 {
     gemdos(X_TABOUT,ch);
 }
+#endif
 
 
 LONG dos_rawcin(void)
@@ -139,10 +141,12 @@ WORD dos_conis(void)
 }
 
 
+#if CONF_WITH_SHOW_FILE
 void dos_conws(const char *str)
 {
     gemdos(X_CONWS,str);
 }
+#endif
 
 
 WORD dos_gdrv(void)
-- 
2.8.2.windows.1

