From 412b771195aaeef7b60dcc24d911fc82eb66c338 Mon Sep 17 00:00:00 2001
From: Christian Zietz <czietz@gmx.net>
Date: Sat, 22 Apr 2017 20:01:55 +0100
Subject: [PATCH] Evaluate Ctrl key at warm boot as well to skip loading AUTO
 programs, accessories, etc.

---
 bios/bios.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/bios/bios.c b/bios/bios.c
index 076298f..d823a2f 100644
--- a/bios/bios.c
+++ b/bios/bios.c
@@ -627,8 +627,12 @@ void biosmain(void)
     show_initinfo = FIRST_BOOT;
 #endif
 
-    if (show_initinfo)
+    if (show_initinfo) {
         bootdev = initinfo();   /* show the welcome screen */
+    } else {
+        /* Allow user to skip AUTO, ACCs, etc. on warm boot. On cold boot this is handled by initinfo(). */
+        bootflags |= (kbshift(-1) & MODE_CTRL)?BOOTFLAG_SKIP_AUTO_ACC:0;
+    }
 
     KDEBUG(("bootdev = %d\n", bootdev));
 
-- 
2.8.2.windows.1

