diff --git a/util/boot.c b/util/boot.c
index 1e23055e..b1a39b1b 100644
--- a/util/boot.c
+++ b/util/boot.c
@@ -118,7 +118,7 @@ int main(void)
 
   /* supervisor */
 
-  Super(0);
+  if (!Super(1)) Super(0);
 
   cpu = get_cpu_cookie();
 
diff --git a/util/minicrt.S b/util/minicrt.S
index 2fd78947..b2acaa63 100644
--- a/util/minicrt.S
+++ b/util/minicrt.S
@@ -34,6 +34,13 @@
         .text
 
 _start:
+        bra.w   prgstart
+        lea.l   -0x20(pc),a0
+        lea.l   stack_base,sp    // set up local stack
+        move.l  a0,-(sp)         // and build initial stack frame
+        bra.s   sysstart
+
+prgstart:
         move.l  4(sp),a0         // pick up base page address
         lea.l   stack_base,sp    // set up local stack
         move.l  a0,-(sp)         // and build initial stack frame
@@ -53,7 +60,7 @@ _start:
         add.l   #12,sp
 
 // empty argc and argv
-
+sysstart:
         moveq   #0,d0
         move.l  d0,-(sp)         // empty env
         move.l  d0,-(sp)         // empty argv
