diff --git a/bios/screen.c b/bios/screen.c
index 007d3aeb..d7ab6f68 100644
--- a/bios/screen.c
+++ b/bios/screen.c
@@ -446,7 +446,9 @@ static void screen_init_mode(void)
 #endif
     WORD monitor_type, sync_mode;
     WORD rez = 0;   /* avoid 'may be uninitialized' warning */
-
+    volatile BYTE trigger;
+    MAYBE_UNUSED(trigger);
+    
     /* Initialize the interrupt handlers & the VBL semaphore.
      * It is important to do this first because the initialization code below
      * may call vsync(), which temporarily enables the interrupts. */
@@ -523,8 +525,11 @@ static void screen_init_mode(void)
         /* On ST, it is important to change the resolution register when nothing
          * is displaying, otherwise the plane shift bug may appear. */
         vsync();
+        vsync();
 
         rez = monitor_type?ST_LOW:ST_HIGH;
+        
+        trigger = *(volatile BYTE*)0xFB0000L;
         *(volatile BYTE *) ST_SHIFTER = rez;
     }
 
diff --git a/bios/startup.S b/bios/startup.S
index b8d34e12..b06fe2a7 100644
--- a/bios/startup.S
+++ b/bios/startup.S
@@ -64,7 +64,7 @@
         .globl  _run
 
 #if DIAGNOSTIC_CARTRIDGE
-    .dc.l   0xfa52235f  // diagnostic cartridge magic number
+    //.dc.l   0xfa52235f  // diagnostic cartridge magic number
 #endif
 
 #if defined(MACHINE_AMIGA) && !EMUTOS_LIVES_IN_RAM
diff --git a/emutos.ld b/emutos.ld
index 076a4d4d..11d98329 100644
--- a/emutos.ld
+++ b/emutos.ld
@@ -21,7 +21,7 @@ OUTPUT_FORMAT(binary)
 #if defined (MACHINE_AMIGA)
 # define ROM_ORIGIN 0x00fc0000
 #elif defined (TARGET_CART)
-# define ROM_ORIGIN 0x00fa0000
+# define ROM_ORIGIN 0x00e00000
 #elif defined (TARGET_192)
 # define ROM_ORIGIN 0x00fc0000
 #else
