From aac27ccfaa99b71e4826b59cb55ae14515508d72 Mon Sep 17 00:00:00 2001
From: Christian Zietz <czietz@gmx.net>
Date: Sun, 8 Oct 2017 10:16:44 +0100
Subject: [PATCH] Read keyboard ACIA data register at boot to clear any pending
 interrupts

---
 bios/aciavecs.S | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/bios/aciavecs.S b/bios/aciavecs.S
index f4e60d9..885d315 100644
--- a/bios/aciavecs.S
+++ b/bios/aciavecs.S
@@ -205,6 +205,13 @@ _init_acia_vecs:
         move.b  d0,joybuf+1
         move.b  d0,joybuf+2
 
+#if CONF_WITH_IKBD_ACIA
+        // clear any pending data from IKBD ACIA
+        .equ    ikbd_acia_stat, 0xfffffc00
+        .equ    ikbd_acia_data, 0xfffffc02
+        move.b  ikbd_acia_data,d0
+#endif
+
 #if CONF_WITH_IKBD_ACIA || CONF_WITH_MIDI_ACIA
         // finally, setup the interrupt vector
         pea     _int_acia
@@ -361,8 +368,6 @@ just_rts:
         rts
 
 #if CONF_WITH_IKBD_ACIA
-        .equ    ikbd_acia_stat, 0xfffffc00
-        .equ    ikbd_acia_data, 0xfffffc02
 
 _ikbdsys:
         moveq   #0,d0
-- 
2.8.2.windows.1

