From 463d7f7ca1cb0ca6771a2fad21268898a086d33a Mon Sep 17 00:00:00 2001
From: Christian Zietz <czietz@gmx.net>
Date: Sat, 23 Sep 2017 15:22:09 +0100
Subject: [PATCH] Open desktop window for empty drives again

---
 desk/deskfpd.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/desk/deskfpd.c b/desk/deskfpd.c
index 1612265..fed4292 100644
--- a/desk/deskfpd.c
+++ b/desk/deskfpd.c
@@ -228,7 +228,7 @@ FNODE *pn_sort(PNODE *pn)
  *  Build the filenode list for the specified pathnode
  *
  *  returns 0   0 or more files found without error
- *          <0  error (other than ENMFIL) returned by dos_sfirst()/dos_snext()
+ *          <0  error (other than EFILNF/ENMFIL) returned by dos_sfirst()/dos_snext()
  *              (e.g. when attempting to open a floppy drive with no disk)
  */
 WORD pn_active(PNODE *pn)
@@ -296,5 +296,5 @@ WORD pn_active(PNODE *pn)
         pn->p_flist = pn_sort(pn);
     }
 
-    return (ret==ENMFIL) ? 0 : ret;
+    return ((ret==ENMFIL) || (ret==EFILNF)) ? 0 : ret;
 }
-- 
2.8.2.windows.1

