From 214bc6504e4c10d0ae4a55e92da7ea0150cb7db3 Mon Sep 17 00:00:00 2001
From: Christian Zietz <czietz@gmx.net>
Date: Sun, 16 Apr 2017 22:07:31 +0100
Subject: [PATCH] Disable 'Install application' for desktop shortcuts. Was
 causing a crash before.

---
 desk/deskmain.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/desk/deskmain.c b/desk/deskmain.c
index f6f185d..0cb384b 100644
--- a/desk/deskmain.c
+++ b/desk/deskmain.c
@@ -276,7 +276,8 @@ static void men_update(void)
         switch(appl->a_type)
         {
         case AT_ISFILE:
-            if (isapp || is_installed(appl))
+            /* do not enable "install application" for desktop shortcuts */
+            if ((!(appl->a_flags & AF_ISDESK)) && (isapp || is_installed(appl)))
                 pvalue = ILL_FILE;
             else
                 pvalue = ILL_DOCU;
-- 
2.8.2.windows.1

