From c0b684e93bbb4189992e0330537095dc64f1997e Mon Sep 17 00:00:00 2001
From: Christian Zietz <czietz@gmx.net>
Date: Sun, 16 Apr 2017 22:35:33 +0100
Subject: [PATCH] Fix 'Install application' for desktop shortcuts

---
 desk/deskins.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/desk/deskins.c b/desk/deskins.c
index bed8f05..707f38f 100644
--- a/desk/deskins.c
+++ b/desk/deskins.c
@@ -363,10 +363,21 @@ WORD ins_app(WORD curr)
     {
         if (!isapp)     /* selected item appears to be a data file */
             return 0;
+#if CONF_WITH_DESKTOP_SHORTCUTS
+        /* Special handling for desktop shortcuts */
+        if (pa->a_flags & AF_ISDESK) {
+            p = pa->a_pdata;
+            q = filename_start(p);
+            pfname = pa->a_pappl;
+        } else {
+#endif
         pw = win_find(G.g_cwin);
         p = pw->w_path->p_spec;
         q = filename_start(p);
         pfname = pf->f_name;
+#if CONF_WITH_DESKTOP_SHORTCUTS
+        }
+#endif
     }
     strlcpy(pathname,p,q-p+1);  /* copy pathname including trailing backslash */
 
-- 
2.8.2.windows.1

