diff --git a/vdi/vdi_control.c b/vdi/vdi_control.c
index ae6a9d3..fb1056e 100644
--- a/vdi/vdi_control.c
+++ b/vdi/vdi_control.c
@@ -328,6 +328,12 @@ void vdi_v_opnvwk(Vwk * vwk)
     WORD handle;
     Vwk **p;
 
+    /*
+     * ensure that CUR_WORK always points to a valid workstation
+     * even if v_opnvwk() exits early.
+     */
+    CUR_WORK = &phys_work;
+
     /* First find a free handle */
     for (handle = VDI_PHYS_HANDLE+1, p = vwk_ptr+handle; handle <= LAST_VDI_HANDLE; handle++, p++) {
         if (!*p) {
diff --git a/vdi/vdi_main.c b/vdi/vdi_main.c
index 2cf3753..eabd803 100644
--- a/vdi/vdi_main.c
+++ b/vdi/vdi_main.c
@@ -147,6 +147,17 @@ void screen(void)
     }
 
     /*
+     * at this point, for v_opnwk() and v_opnvwk(), vwk is NULL.  we
+     * must fix this before we use it to set the lineA variables below.
+     * fortunately, v_opnwk() and v_opnvwk() have set CUR_WORK to a
+     * valid value (see vdi_control.c).  so we use this to set vwk.
+     */
+    if ((opcode == 1) || (opcode == 100))
+    {
+        vwk = CUR_WORK;
+    }
+
+    /*
      * set some line-A variables from the vwk info (as long as
      * the workstation is valid)
      */
