From 0a07c4d7d6374794d7b30eb12f814fb6182f46b5 Mon Sep 17 00:00:00 2001
From: Christian Zietz <czietz@gmx.net>
Date: Fri, 7 Apr 2017 21:28:42 +0100
Subject: [PATCH] Fix AES dispatching to next process right after APPL_INIT

---
 aes/gemsuper.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/aes/gemsuper.c b/aes/gemsuper.c
index 078650c..95930aa 100644
--- a/aes/gemsuper.c
+++ b/aes/gemsuper.c
@@ -494,7 +494,7 @@ WORD super(WORD cx, AESPB *pcrys_blk)
     if (cx == 200)
         xif(pcrys_blk);
 
-    if ((dspcnt++ % 8) == 0 || cx == 201)
+    if ((++dspcnt % 8) == 0 || cx == 201)
         dsptch();
 
     return 0;
-- 
2.8.2.windows.1

