diff --git a/bios/startup.S b/bios/startup.S
index f655144e..0fd6de9b 100644
--- a/bios/startup.S
+++ b/bios/startup.S
@@ -304,10 +304,11 @@ cache_done:
  */
 #ifndef __mcoldfire__
         move.l  #not_68030,0x2c    // PMOVE is lineF on non-68030 systems
-        PMOVE_TO_TTR0(zero)        // first we initialise the TTRs (ACRs on a 68ec030)
-        PMOVE_TO_TTR1(zero)
+        lea     zero,a0
+        PMOVE_A0_TTR0              // first we initialise the TTRs (ACRs on a 68ec030)
+        PMOVE_A0_TTR1
         move.l  #mmu_done,0x2c     // since PMOVE_TO_TC doesn't exist on a 68ec030,
-        PMOVE_TO_TC(zero)          //  we're done if we get a lineF exception ...
+        PMOVE_A0_TC                //  we're done if we get a lineF exception ...
         jra     mmu_done
 not_68030:
         move.l  #mmu_done,0x10     // if a MOVEC causes an exception, we're done
diff --git a/include/asmdefs.h b/include/asmdefs.h
index ec9d93ac..c215902c 100644
--- a/include/asmdefs.h
+++ b/include/asmdefs.h
@@ -51,6 +51,10 @@
 #define PMOVE_TO_TTR0(addr) .dc.l 0xf0390800,addr   /* 68030 */
 #define PMOVE_TO_TTR1(addr) .dc.l 0xf0390c00,addr   /* 68030 */
 
+#define PMOVE_A0_TC         .dc.l 0xf0104000        /* 68030 (except 68ec030) */
+#define PMOVE_A0_TTR0       .dc.l 0xf0100800        /* 68030 */
+#define PMOVE_A0_TTR1       .dc.l 0xf0100c00        /* 68030 */
+
 #define FNOP                .dc.l 0xf2800000        /* 6888X, 68040-68060 (except 68ec040/68ec060) */
 #define FSAVE_MINUS_SP      .dc.w 0xf327            /* 6888X, 68040-68060 (except 68ec040/68ec060) */
 #define FRESTORE_SP_PLUS    .dc.w 0xf35f            /* 6888X, 68040-68060 (except 68ec040/68ec060) */
