diff --git a/include/asm.h b/include/asm.h
index 3c1bf96e..da13ba4f 100644
--- a/include/asm.h
+++ b/include/asm.h
@@ -302,17 +302,19 @@ __extension__                             \
 #define regsafe_call(addr)                         \
 __extension__                                      \
 ({__asm__ volatile ("lea     -60(sp),sp\n\t"       \
-                    "movem.l d0-d7/a0-a6,(sp)");   \
-  ((void (*)(void))addr)();                        \
-  __asm__ volatile ("movem.l (sp),d0-d7/a0-a6\n\t" \
-                    "lea     60(sp),sp");          \
+                    "movem.l d0-d7/a0-a6,(sp)\n"   \
+                    "jsr (%0)\n"                   \
+                    "movem.l (sp),d0-d7/a0-a6\n\t" \
+                    "lea     60(sp),sp\n"          \
+                    : : "a"(addr));  \
 })
 #else
 #define regsafe_call(addr)                         \
 __extension__                                      \
-({__asm__ volatile ("movem.l d0-d7/a0-a6,-(sp)");  \
-  ((void (*)(void))addr)();                        \
-  __asm__ volatile ("movem.l (sp)+,d0-d7/a0-a6");  \
+({__asm__ volatile ("movem.l d0-d7/a0-a6,-(sp)\n"  \
+                    "jsr (%0)\n"                   \
+                    "movem.l (sp)+,d0-d7/a0-a6\n"  \
+                    : : "a"(addr));  \
 })
 #endif
 
