diff --git a/bios/screen.c b/bios/screen.c
index 007d3aeb..3cc07d28 100644
--- a/bios/screen.c
+++ b/bios/screen.c
@@ -446,7 +446,9 @@ static void screen_init_mode(void)
 #endif
     WORD monitor_type, sync_mode;
     WORD rez = 0;   /* avoid 'may be uninitialized' warning */
-
+    volatile BYTE trigger;
+    MAYBE_UNUSED(trigger);
+    
     /* Initialize the interrupt handlers & the VBL semaphore.
      * It is important to do this first because the initialization code below
      * may call vsync(), which temporarily enables the interrupts. */
@@ -525,6 +527,8 @@ static void screen_init_mode(void)
         vsync();
 
         rez = monitor_type?ST_LOW:ST_HIGH;
+        
+        trigger = *(volatile BYTE*)0xFB0000L;
         *(volatile BYTE *) ST_SHIFTER = rez;
     }
 
diff --git a/bios/startup.S b/bios/startup.S
index b8d34e12..9f5dac56 100644
--- a/bios/startup.S
+++ b/bios/startup.S
@@ -241,6 +241,16 @@ not_falcon:
 main_after_reset_instruction:
 #endif /* CONF_WITH_RESET && !defined(MACHINE_AMIGA) */
 
+
+        // CHZ: we want to do a reset here for debugging purposes which messes with MMU as well
+        move.b  0xffff8001.w, d1
+        lea     0x380, a0
+        move.w  #0x4e70, (a0)+       // RESET
+        move.l  #0x11c18001, (a0)+   // MOVE.B D1, 0xFFFF8001.w
+        move.w  #0x4e75, (a0)+       // RTS
+        jsr     0x380
+
+
 #ifdef __mcoldfire__
         lea     coldfire_startup_ret,a6
         jmp     coldfire_startup
