From 077e4f1de1bc0dec459bd849c20a8b909e9e96ae Mon Sep 17 00:00:00 2001
From: Christian Zietz <czietz@gmx.net>
Date: Sun, 8 Oct 2017 13:04:52 +0100
Subject: [PATCH] Retry identifying harddisks at boot

---
 bios/disk.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/bios/disk.c b/bios/disk.c
index 131dbe1..ae82d0c 100644
--- a/bios/disk.c
+++ b/bios/disk.c
@@ -21,6 +21,7 @@
 #include "kprint.h"
 #include "xhdi.h"
 #include "processor.h"
+#include "machine.h"
 #include "natfeat.h"
 #include "tosvars.h"
 #include "mfp.h"
@@ -91,6 +92,10 @@ static void disk_init_one(UWORD unit,LONG *devices_available)
     {
         /* Try our internal drivers */
         rc = internal_inquire(unit, NULL, &device_flags, productname, sizeof productname);
+        if (!(IS_ARANYM) && rc) {
+            /* Retry once */
+            rc = internal_inquire(unit, NULL, &device_flags, productname, sizeof productname);
+        }
         if (rc) {
             KDEBUG(("disk_init_one(): internal_inquire(%d) returned %ld\n",unit,rc));
             return;
-- 
2.8.2.windows.1

