diff --git a/bios/bios.c b/bios/bios.c
index 9ddd4c9..44ea668 100644
--- a/bios/bios.c
+++ b/bios/bios.c
@@ -420,8 +420,7 @@ static void bios_init(void)
     if (has_nova && !(kbshift(-1) & MODE_CTRL)) {
         KDEBUG(("init_nova()\n"));
         if (init_nova()) {
-            set_rez_hacked();
-            vt52_init();            /* initialize the vt52 console */
+            set_rez_hacked();   /* also reinitializes the vt52 console */
         }
     }
 #endif
@@ -471,9 +470,7 @@ static void bios_init(void)
 
         if ((V_REZ_HZ != save_hz) || (V_REZ_VT != save_vt) || (v_planes != save_pl))
         {
-            set_rez_hacked();
-            set_screen_shift();     /* set shift amount for screen address calc */
-            vt52_init();            /* initialize the vt52 console */
+            set_rez_hacked();   /* also reinitializes the vt52 console */
         }
     }
 #endif
diff --git a/bios/nova.c b/bios/nova.c
index 7c4a032..337c2b4 100644
--- a/bios/nova.c
+++ b/bios/nova.c
@@ -531,8 +531,6 @@ static void init_system_vars(void)
     /* Line A vars */
     /* Number of bitplanes */
     v_planes = 1;
-    /* precalculate shift value to optimize pixel address calculations */
-    v_planes_shift = 3;     /* see shift_offset[] in lineainit.c */
     /* Bytes per scan-line */
     BYTES_LIN = v_lin_wr = 80;
     /* Vertical resolution */
diff --git a/bios/screen.c b/bios/screen.c
index 57ea679..a59b099 100644
--- a/bios/screen.c
+++ b/bios/screen.c
@@ -625,12 +625,15 @@ void screen_init_address(void)
 /*
  * Mark resolution as hacked
  *
- * called by bios_init() if a cartridge application has altered key
- * lineA variables
+ * called by bios_init() if a special video mode (Nova support, Hatari
+ * cartridge extended VDI) has altered key lineA variables
  */
 void set_rez_hacked(void)
 {
     rez_was_hacked = TRUE;
+
+    set_screen_shift();     /* set shift amount for screen address calc */
+    vt52_init();            /* initialize the vt52 console */
 }
 
 /*
