diff --git a/bdos/bdosmain.c b/bdos/bdosmain.c
index d4fb360..7581246 100644
--- a/bdos/bdosmain.c
+++ b/bdos/bdosmain.c
@@ -283,9 +283,10 @@ void    osinit(void)
 
     run->p_uft[0] = H_Console;          /* stdin        =       con:    */
     run->p_uft[1] = H_Console;          /* stdout       =       con:    */
-    run->p_uft[2] = H_Console;          /* stderr       =       con:    */
-    run->p_uft[3] = H_Aux;              /* stdaux       =       aux:    */
-    run->p_uft[4] = H_Print;            /* stdprn       =       prn:    */
+    run->p_uft[2] = H_Aux;              /* stdaux       =       aux:    */
+    run->p_uft[3] = H_Print;            /* stdprn       =       prn:    */
+    run->p_uft[4] = H_Console;          /* Atari TOS redirects undefined ... */
+    run->p_uft[5] = H_Console;          /* ... handles 4, 5 to con: */
 
     add[0] = remove[0] = add[1] = remove[1] = add[2] = remove[2] = 0 ;
 
diff --git a/bdos/console.c b/bdos/console.c
index af491aa..06d7983 100644
--- a/bdos/console.c
+++ b/bdos/console.c
@@ -100,7 +100,7 @@ long xconostat(void)
  */
 long xprtostat(void)
 {
-    return(Bcostat(HXFORM(run->p_uft[4])));
+    return(Bcostat(HXFORM(run->p_uft[3])));
 }
 
 /*
@@ -108,7 +108,7 @@ long xprtostat(void)
  */
 long xauxistat(void)
 {
-    return(constat(HXFORM(run->p_uft[3])));
+    return(constat(HXFORM(run->p_uft[2])));
 }
 
 /*
@@ -116,7 +116,7 @@ long xauxistat(void)
  */
 long xauxostat(void)
 {
-    return(Bcostat(HXFORM(run->p_uft[3])));
+    return(Bcostat(HXFORM(run->p_uft[2])));
 }
 
 /*
@@ -244,7 +244,7 @@ static void cookdout(int h, int ch)
  */
 long xauxout(int ch)
 {
-    return Bconout(HXFORM(run->p_uft[3]), ch);
+    return Bconout(HXFORM(run->p_uft[2]), ch);
 }
 
 /*
@@ -252,7 +252,7 @@ long xauxout(int ch)
  */
 long xprtout(int ch)
 {
-    return Bconout(HXFORM(run->p_uft[4]), ch);
+    return Bconout(HXFORM(run->p_uft[3]), ch);
 }
 
 
@@ -334,7 +334,7 @@ long x8in(void)
  */
 long xauxin(void)
 {
-    return(Bconin(HXFORM(run->p_uft[3])));
+    return(Bconin(HXFORM(run->p_uft[2])));
 }
 
 /*
