diff --git a/bios/memory2.c b/bios/memory2.c
index 9db0abb0..3039ca59 100644
--- a/bios/memory2.c
+++ b/bios/memory2.c
@@ -36,7 +36,7 @@ static ULONG detect_ttram_size(void)
     /* Fixed-size TT-RAM */
     return CONF_TTRAM_SIZE;
 #else
-    UBYTE *addr;
+    volatile UBYTE *addr;
 
     /* By design, TT-RAM is always in 32-bit address space */
     if (!IS_BUS32)
@@ -53,8 +53,15 @@ static ULONG detect_ttram_size(void)
     for (addr = TTRAM_START + 1024UL*1024 - 1;
         (long)addr > 0; addr += 1024UL*1024)
     {
+        /* First check for bus error, then try writing to memory and
+         * reading back. */
         if (!check_read_byte((long)addr))
             break;
+
+        *addr     = 0x12;
+        *(addr+1) = 0x34;
+        if ((*addr != 0x12) || (*(addr+1) != 0x34))
+            break;
     }
 
     /* Valid TT-RAM stops at the beginning of current megabyte */
