diff --git a/aes/gemevlib.c b/aes/gemevlib.c
index 6e77be68..25a3f902 100644
--- a/aes/gemevlib.c
+++ b/aes/gemevlib.c
@@ -132,7 +132,7 @@ void ev_mouse(MOBLK *pmo, WORD rets[])
  */
 void ev_timer(LONG count)
 {
-    ev_block(MU_TIMER, count/gl_ticktime);
+    ev_block(MU_TIMER, (ULONG)count/gl_ticktime);
 }
 
 
@@ -234,7 +234,7 @@ WORD ev_multi(WORD flags, MOBLK *pmo1, MOBLK *pmo2, LONG tmcount,
         }
         /* wait for timer */
         if (flags & MU_TIMER)
-            iasync(MU_TIMER, tmcount/gl_ticktime);
+            iasync(MU_TIMER, (ULONG)tmcount/gl_ticktime);
         /* wait for events */
         which = mwait(flags);
 
diff --git a/aes/gemgsxif.c b/aes/gemgsxif.c
index 5264aebd..9199bbd6 100644
--- a/aes/gemgsxif.c
+++ b/aes/gemgsxif.c
@@ -352,7 +352,7 @@ static void bb_set(WORD sx, WORD sy, WORD sw, WORD sh, WORD *pts1, WORD *pts2,
 
     if (size > gl_mlen) {       /* buffer too small */
         /* adjust height to fit buffer: this will leave droppings! */
-        sh = gl_mlen * sh / size;
+        sh = (ULONG)gl_mlen * sh / size;
 
         /* issue warning message for backup only, not for subsequent restore */
         if (pdst == &gl_tmp)
