diff --git a/bios/amiga.c b/bios/amiga.c
index 398899a7..773ddffd 100644
--- a/bios/amiga.c
+++ b/bios/amiga.c
@@ -620,6 +620,7 @@ void amiga_screen_init(void)
      */
 
     /* Set up the Copper list (must be in ST-RAM) */
+    (void)balloc_stram(116, FALSE);
     copper_list = (UWORD *)balloc_stram(sizeof(UWORD) * 8, FALSE);
     copper_list[0] = 0x0a01; /* Wait line 10 to give time to the VBL routine */
     copper_list[1] = 0xff00; /* Vertical wait only */
