diff --git a/vdi/vdi_col.c b/vdi/vdi_col.c
index 3c7419d..dc6228a 100644
--- a/vdi/vdi_col.c
+++ b/vdi/vdi_col.c
@@ -215,7 +215,7 @@ static const WORD videl_palette2[240][3] =
 /* Create an ST color value from VDI color */
 static int vdi2st(int col)
 {
-    return col * 7 / 1000;
+    return (col + 72) / 143;
 }
 
 
@@ -236,7 +236,7 @@ static const WORD st2vdi_lookup_table[8] =
 /* Create an STe color value from VDI color */
 static int vdi2ste(int col)
 {
-    col = col * 3 / 200;
+    col = (col * 3 + 100) / 200;
     col = ((col & 1) << 3) | (col >> 1);  /* Shift lowest bit to top */
 
     return col;
