diff --git a/bios/cookie.h b/bios/cookie.h
index 38909d6..132b0ef 100644
--- a/bios/cookie.h
+++ b/bios/cookie.h
@@ -42,6 +42,8 @@ struct cookie {
 #define COOKIE_COLDFIRE 0x5f43465fL
 #define COOKIE_MCF      0x5f4d4346L
 
+#define COOKIE_EMUVER   0x456d7554L
+
 /*
  * values of _MCH cookie
  */
diff --git a/bios/machine.c b/bios/machine.c
index 64f7ee5..741fb3e 100644
--- a/bios/machine.c
+++ b/bios/machine.c
@@ -34,6 +34,7 @@
 #include "mfp.h"
 #include "scc.h"
 #include "coldfire.h"
+#include "version.h"
 #ifdef MACHINE_AMIGA
 #include "amiga.h"
 #endif
@@ -606,6 +607,9 @@ void fill_cookie_jar(void)
 #if CONF_WITH_XHDI
     create_XHDI_cookie();
 #endif
+
+    /* EmuTOS version cookie */
+    cookie_add(COOKIE_EMUVER, (long)&version);
 }
 
 static const char * guess_machine_name(void)
