diff --git a/aes/geminit.c b/aes/geminit.c
index 02171f0..723cfbf 100644
--- a/aes/geminit.c
+++ b/aes/geminit.c
@@ -419,6 +419,39 @@ void all_run(void)
 
 
 /*
+ * this waits until all the desk accessories are waiting for a message.
+ * it is called after the accessories have been loaded, as a way of
+ * ensuring they have initialised before starting the desktop (or an
+ * autorun program, if present).
+ */
+static void wait_for_accs(void)
+{
+    AESPD *pd;
+    WORD pid;
+
+    while(1)
+    {
+        /* let everybody run */
+        all_run();
+
+        /*
+         * check the EVB lists for all the DAs (pid 0 & 1 are system)
+         */
+        for (pid = 2; ; pid++)
+        {
+            pd = fpdnm(NULL, pid);
+            if (!pd)        /* no more DAs: they must all */
+                return;     /*  be waiting for messages   */
+
+            KDEBUG(("%s wait=0x%04x\n",pd->p_name,pd->p_evwait));
+            if (!(pd->p_evwait&MU_MESAG))
+                break;      /* force us to go round again */
+        }
+    }
+}
+
+
+/*
  *  This function is called from accdesk_start (in gemstart.S) which
  *  is itself called from gem_main() below.
  *
@@ -477,7 +510,7 @@ void run_accs_and_desktop(void)
     isgem = process_inf2();         /* process emudesk.inf part 2 */
 
     dsptch();                       /* off we go !!! */
-    all_run();                      /* let them run  */
+    wait_for_accs();                /* wait until DAs have initialised */
 
     sh_init();                      /* init for shell loop */
     sh_main(isgem);                 /* main shell loop */
