diff --git a/cli/cmd.h b/cli/cmd.h
index 74c52706..006f2aea 100644
--- a/cli/cmd.h
+++ b/cli/cmd.h
@@ -82,6 +82,7 @@ extern LONG jmp_xbios(WORD, ...);
 
 #define Getrez()            jmp_xbios_v(0x04)
 #define Setscreen(a,b,c,d)  jmp_xbios_llww(0x05,a,b,c,d)
+#define Setcolor(a,b)       jmp_xbios_ww(0x07,a,b)
 #define Cursconf(a,b)       jmp_xbios_ww(0x15,a,b)
 #define Kbrate(a,b)         jmp_xbios_ww(0x23,a,b)
 #define Supexec(a)          jmp_xbios_l(0x26,a)
diff --git a/cli/cmdmain.c b/cli/cmdmain.c
index 4849b5c1..cf9fad25 100644
--- a/cli/cmdmain.c
+++ b/cli/cmdmain.c
@@ -215,14 +215,17 @@ PRIVATE void change_res(WORD res)
 #else
     /* mode changed *without* palette change -> set readable text color index */
     {
-        /* OS masks color index, so 15 is fine also for mono modes */
-        int idx = 15;
-        /* from first 4 entries in LOW palette, red is better than yellow */
+        static int old_color_3 = -1;
+        /* switching to ST medium: set color 3 to black */
         if (res == ST_MEDIUM)
-            idx = 1;
+            old_color_3 = Setcolor(3, 0);
+        /* switching from ST medium: reset color 3 */
+        if (current_res == ST_MEDIUM && old_color_3 != -1)
+            Setcolor(3, old_color_3);
         conout(ESC);    /* with VT52 command */
         conout('b');    /* b=foreground, c=background */
-        conout(idx);
+        /* OS masks color index, so 15 is fine also for mono/medium modes */
+        conout(15);
     }
 #endif
     enable_cursor();
