diff --git a/bios/bios.c b/bios/bios.c
index 695e5557..f1e600ec 100644
--- a/bios/bios.c
+++ b/bios/bios.c
@@ -17,7 +17,7 @@
  * option any later version.  See doc/license.txt for details.
  */
 
-/* #define ENABLE_KDEBUG */
+#define ENABLE_KDEBUG
 
 #include "emutos.h"
 #include "biosext.h"
diff --git a/bios/disk.c b/bios/disk.c
index ae90dfae..8fcda41c 100644
--- a/bios/disk.c
+++ b/bios/disk.c
@@ -10,7 +10,7 @@
  * option any later version.  See doc/license.txt for details.
  */
 
-/*#define ENABLE_KDEBUG*/
+#define ENABLE_KDEBUG
 
 #include "emutos.h"
 #include "gemerror.h"
diff --git a/bios/ide.c b/bios/ide.c
index 5f885a29..7aa0434a 100644
--- a/bios/ide.c
+++ b/bios/ide.c
@@ -17,7 +17,7 @@
  * Note: this driver does not support CHS addressing.
  */
 
-/* #define ENABLE_KDEBUG */
+#define ENABLE_KDEBUG
 
 #include "emutos.h"
 #include "asm.h"
diff --git a/include/config.h b/include/config.h
index 9a564624..a467a3a2 100644
--- a/include/config.h
+++ b/include/config.h
@@ -1369,7 +1369,7 @@
  * Set CONSOLE_DEBUG_PRINT to 1 to redirect debug prints to the BIOS console.
  */
 #ifndef CONSOLE_DEBUG_PRINT
-# define CONSOLE_DEBUG_PRINT 0
+# define CONSOLE_DEBUG_PRINT 1
 #endif
 
 /*
