diff --git a/bios/acsi.c b/bios/acsi.c
index 37a7282a..feb85ec4 100644
--- a/bios/acsi.c
+++ b/bios/acsi.c
@@ -348,6 +348,7 @@ static int send_command(UBYTE *inputcdb,WORD cdblen,WORD rw,WORD dev,WORD cnt,UW
      * see if we need to use ICD trickery
      */
     if (*inputcdb > 0x1e) {
+        return -1;
         cdbptr = cdb;
         *cdbptr = 0x1f;     /* ICD extended command method */
         memcpy(cdbptr+1,inputcdb,cdblen);
diff --git a/include/config.h b/include/config.h
index 9db0798e..1f2aa50d 100644
--- a/include/config.h
+++ b/include/config.h
@@ -718,7 +718,7 @@
  * MonSTer expansion card
  */
 #ifndef CONF_WITH_MONSTER
-# define CONF_WITH_MONSTER 1
+# define CONF_WITH_MONSTER 0
 #endif
 
 /*
